src/waveform/renderers/waveformmarkset.cpp

```diff
@@ -69,3 +69,43 @@ WaveformMarkPointer WaveformMarkSet::getHotCueMark(int hotCue) const {
 WaveformMarkPointer WaveformMarkSet::getDefaultMark() const {
     return m_pDefaultMark;
 }
+
+void WaveformMarkSet::update() {
+    std::map<WaveformMarkSortKey, WaveformMarkPointer> map;
+    for (const auto& pMark : m_marks) {
+        if (pMark->isValid() && pMark->isVisible()) {
+            double samplePosition = pMark->getSamplePosition();
+            if (samplePosition != Cue::kNoPosition) {
+                // Create a stable key for sorting, because the WaveformMark's samplePosition is a
+                // ControlObject which can change at any time by other threads. Such a change causes
+                // another updateCues() call, rebuilding map.
+                auto key = WaveformMarkSortKey(samplePosition, pMark->getHotCue());
+                map.emplace(key, pMark);
+            }
+        }
+    }
+
+    m_marksToRender.clear();
+    m_marksToRender.reserve(map.size());
```

Comment:
or maybe I just use a std::vector for m_marksToRender? The use of Qt containers vs std containers throughout the mixxx code seems rather arbitrary. Is there a preference / guideline to follow?

Reply:
The defacto choice has been Qt containers over the years (especially when interacting with Qt APIs since those only take Qt containers and conversion maybe costly). I personally prefer std containers over Qt containers because of their improved ergonomics and better integration into the language. 

I'm surprised we don't mention anything in the coding guideles, but afaik the currently accepted stance is to use Qt containers when interacting with APIs that use them and use your personal preference otherwise.

If you think it makes sense to convert it to a std::vector without introducing more friction, I'd appreciate if you do.

Reply:
no, i will keep it a QList. but instead of using qt version dependent code i prefer casting to QList::size_type.